plan/task/exec.go

```diff
@@ -7,32 +7,75 @@ import (
 	"net"
 	"strings"
 
+	"cuelang.org/go/cue"
 	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/moby/buildkit/solver/pb"
+	"github.com/rs/zerolog/log"
 	"go.dagger.io/dagger/compiler"
+	"go.dagger.io/dagger/pkg"
 	"go.dagger.io/dagger/plancontext"
 	"go.dagger.io/dagger/solver"
 )
 
 func init() {
 	Register("Exec", func() Task { return &execTask{} })
+	Register("AsyncExec", func() Task { return &asyncExecTask{} })
```

Comment:
random suggestion: if it helps avoiding duplicating code, we could do something like:

```go
Register("Exec", func() Task { return &execTask{} })
Register("AsyncExec", func() Task { return &execTask{async: true} })
```

And then use the same task for both sync and async

Reply:
In this case it won't help more as all the deduplication already happens via the `parseCommon` func called by both methods, but that's a good idea in general to take advantage of struct params when helpful